datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1917,6 +1923,84 @@ impl Filter {
 
         Ok(Self { predicate, input })
     }
+
+    fn is_scalar(&self) -> bool {
```

Comment:
Can we please add docstrings that explains what this function does / under what circumstances it returns true?